lib/logstash/outputs/stomp.rb

```diff
@@ -57,11 +66,35 @@ def register
     
     connect
   end # def register
-  
-  def receive(event)
-      
 
-      @logger.debug(["stomp sending event", { :host => @host, :event => event }])
-      @client.send(event.sprintf(@destination), event.to_json)
+  public
+  def do_close
+    @logger.warn("Disconnecting from stomp broker")
+    @client.disconnect if @client.connected?
+  end # def do_close
+
+  def multi_receive(events)
+
+    headers = Hash.new
+    if @headers
+      @headers.each do |k,v|
+        headers[k] = event.sprintf(v)
+      end
+    end
+
+    if @batch
+      @logger.debug(["stomp sending events in batch", { :host => @host, :events => events.length, :headers => headers }])
+
+      @client.transaction do |t|
+        events.each { |event|
+          t.send(event.sprintf(@destination), event.to_json, headers)
+        }
+      end
+    else
+      events.each { |event|
+        @logger.debug(["stomp sending event", { :host => @host, :event => event, :headers => headers }])
+        @client.send(event.sprintf(@destination), event.to_json, headers)
+      }
+    end
```

Comment:
In Logstash `multi_receive` is clear to receive multiple events,  I don't think we should support a batch and a no batch mode.  In all cases the batch mode will be more efficient for Logstash and for stomp itself.

Do you need to enable something special on the server side to enable batch mode? I dont recall that.